src/pytest_cov/engine.py

```diff
@@ -12,6 +14,19 @@
 from .compat import StringIO, workeroutput, workerinput
 
 
+_CovResult = collections.namedtuple("_CovResult", "result,config")
+
+
+def _work_around_config_mutation(cov, method, *args, **kwargs):
```

Comment:
This would be better as a context manager.

Reply:
It won't work as a contextmanager

Reply:
I see: this function also wraps the return in a _CovResult.  This is hard to follow.

Reply:
I would move the _CovResult aspect out of this function, and make it a context manager.

Reply:
This is so there is pure function interface to the cov API. A context manager wouldn't provide that

Reply:
It's also only a temporary workaround, I'll follow up with a PR to coveragepy

Reply:
So this will only work for coverage.py 5.0+?

Reply:
The workaround would version dependent if we wanted to support old coveragepy versions

Reply:
It would work as a context manager, eg:
```py
@contextmanager
def restore(obj, attr):
    backup = getattr(obj, attr)
    try:
        yield
    finally:
        setattr(obj, attr, backup)
```


I think using `with restore(self.cov, 'config'):` would be more clear to the reader about what is actually going on than `_work_around_config_mutation(lots of arguments)`.

Reply:
ok that context manager looks nice @ionelmc